### PR TITLE
XW-4232 | Accessibility of the settings menu

### PR DESCRIPTION
**Pull request by bkoval**
@Wikia/x-wing

### CODE REVIEW

File: src/settings.plugin.js [hunk 1]
```diff
@@ -146,8 +160,10 @@ wikiaJWPlayerSettingsPlugin.prototype.createSubmenuWrapper = function () {
 		submenuWrapper = document.createElement('ul');
 
 	backElement.className = 'wikia-jw-settings__back';
+	backElement.tabIndex = 0;
+	backElement.setAttribute('aria-label', 'Back Button');
```
Comment (latata) [inline src/settings.plugin.js]: use `this.config.i18n.back` instead of hard-coded string

File: src/settings.plugin.js [hunk 2]
```diff
@@ -157,33 +173,52 @@ wikiaJWPlayerSettingsPlugin.prototype.createSubmenuWrapper = function () {
...
+
+			var input = event.target.previousSibling;
+
+			input.checked = !input.checked;
+		}
+	});
+
+	this._onClick(label, toggleAutoplay);
+	label.tabIndex = 0;
+	label.setAttribute('aria-label', 'Autoplay Toggle');
```
Comment (latata) [inline src/settings.plugin.js]: i18n

File: src/settings.plugin.js [hunk 3]
```diff
@@ -157,33 +173,52 @@ wikiaJWPlayerSettingsPlugin.prototype.createSubmenuWrapper = function () {
...
+	var autoplayToggle = createToggle({
+		id: this.player.getContainer().id + '-videoAutoplayToggle',
+		label: this.config.i18n.autoplayVideos,
+		checked: this.config.autoplay
+	});
+
+	var label = autoplayToggle.querySelector('label');
+
+	label.addEventListener('keydown', function (evt) {
+		if ([13, 32].indexOf(evt.keyCode) !== -1) {
```
Comment (latata) [inline src/settings.plugin.js]: Maybe we could extract `[13, 32]` to constant?

File: src/settings.plugin.js [hunk 4]
```diff
@@ -157,33 +173,52 @@ wikiaJWPlayerSettingsPlugin.prototype.createSubmenuWrapper = function () {
...
+			evt.preventDefault();
+			evt.stopPropagation();
+
+			var input = event.target.previousSibling;
+
+			input.checked = !input.checked;
+		}
+	});
+
+	this._onClick(label, toggleAutoplay);
```
Comment (latata) [inline src/settings.plugin.js]: It adds second `keydown` listener. Do we need both?

File: src/settings.plugin.js [hunk 5]
```diff
@@ -157,33 +173,52 @@ wikiaJWPlayerSettingsPlugin.prototype.createSubmenuWrapper = function () {
...
 wikiaJWPlayerSettingsPlugin.prototype.createQualityButton = function () {
+	function openQualityMenu() {
+		hideElement(this.settingsList);
+		showElement(this.qualityLevelsList);
+	}
 	var qualityElement = document.createElement('li');
 
 	qualityElement.className = 'wikia-jw-settings__quality-button';
+	qualityElement.tabIndex = 0;
+	qualityElement.setAttribute('aria-label', 'Quality Menu');
```
Comment (latata) [inline src/settings.plugin.js]: i18n

File: src/settings.plugin.js [hunk 6]
```diff
@@ -220,11 +255,13 @@ wikiaJWPlayerSettingsPlugin.prototype.updateQualityLevelsList = function (newLev
 
 	newLevels.forEach(function (level, index) {
 		var qualityLevelItem = document.createElement('li');
+		qualityLevelItem.tabIndex = 0;
+		qualityLevelItem.setAttribute('aria-label', 'Quality Level ' + level.label);
```
Comment (latata) [inline src/settings.plugin.js]: i18n

File: src/settings.plugin.js (hunk 4 above)
Comment (bkoval) [reply src/settings.plugin.js]: Good point!

**Pull request closed without merging**